examples/step-3/step-3.cc

```diff
@@ -623,32 +623,55 @@ void Step3::run ()
+// when running this tutorial program.
 //
-// The output of deallog can be redirected to the console, to a file, or
-// both. But both are disabled by default. The output is nested in a way so
-// that each function can use a prefix string (separated by colons) for each
-// line of output; if it calls another function, that may also use its prefix
-// which is then printed after the one of the calling function. By running
-// this example (or looking at the "Results" section), you will see the solver
-// statistics prefixed with "DEAL:CG", which is two prefixes.  Since output
-// from functions which are nested deep below is usually not as important as
-// top-level output, you can give the deallog variable a maximal depth of
-// nested output for output to console and file. A depth of 0 (the default)
-// will disable output to the screen, while a value of 2 or higher will cause
-// the solver info in this example to be printed. Imagine that different
-// solvers can be nested, which we will see in step-22 for example, and you
-// might not want to see all this information.
+// The output of @p deallog can be written to the console, to a file,
+// or both. Both are disabled by default since over the years we have
+// learned that a program should only generate output when a user
+// explicitly asks for it. But this can be changed, and to explain how
+// this can be done, we need to explain how @p deallog works: When
+// individual parts of the library want to log output, they open a
+// "context" or "section" into which this output will be placed. At
+// the end of the part that wants to write output, one exits this
+// section again. Since a function may call another one from within
+// the scope where this output section is open, output may in fact be
+// nested hierarchically into these sections. The LogStream class of
+// which @p deallog is a variable calls each of these sections a
+// "prefix" because all output is printed with this prefix at the left
+// end of the line, with prefixes separated by colons. There is always
+// a default prefix called "DEAL" (a hint at deal.II's history as the
+// successor of a previous library called "DEAL" and from which the
+// LogStream class is one of the few pieces of code that were taken
+// into deal.II).
+//
+// By default, @p logstream only outputs lines with zero prefixes --
+// i.e., all output is disabled because the default "DEAL" prefix is
+// always there. But one can set a different maximal number of
+// prefixes for lines that should be output to something larger, and
+// indeed here we set it to two by calling
+// LogStream::depth_console(). This means that for all screen output,
+// a context that has pushed one additional prefix beyond the default
+// "DEAL" is allowed to print its output to the screen ("console"),
+// whereas all further nested sections that would have three or more
+// prefixes active would simply have their output go to nirvana. Thus,
```

Comment:
In the end, we simply omit writing anywhere in this case. Therefore, I doubt that "nirvana" is precise here. :wink:

Reply:
Fair enough -- I updated the text here :-)